src/class/hid/hid_device.c

```diff
@@ -192,6 +193,10 @@ bool hidd_open(uint8_t rhport, tusb_desc_interface_t const * desc_itf, uint16_t
 // return false to stall control endpoint (e.g unsupported request)
 bool hidd_control_request(uint8_t rhport, tusb_control_request_t const * p_request)
 {
+  if (p_request->bmRequestType_bit.recipient != TUSB_REQ_RCPT_INTERFACE)

```

Comment:
the recipient can be device in some cases, please revert this

Reply:
Something still seems wrong. The next lines try to lookup the interface based on wIndex. If it's a device target, that won't be right. Of the top of my head, I'm not sure what wIndex means for device requests.

I'll revert this, and let you figure out the proper checks?